### PR TITLE
Add option to check for deprecated stacks in pre-start rake task

### PR DESCRIPTION
**Pull request by xandroc**
* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [ ] I have run CF Acceptance Tests on bosh lite

### CODE REVIEW

File: jobs/cloud_controller_ng/spec [hunk 1]
```diff
@@ -467,6 +469,10 @@ properties:
   cc.default_stack:
     default: "cflinuxfs4"
     description: "The default stack to use if no custom stack is specified for an app."
+  cc.deprecated_stacks:
+    description: |
+      List of deprecated/unsupported stack names. If a stack name is in this list, Cloud Controller bootstrap VM
+      will fail to start up if the stack is found in the database but is not explicitly listed in the stacks property.
```
Comment (joaopapereira) [inline jobs/cloud_controller_ng/spec:475]: This comment is a little hard to understand. Can we rephrase it?

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Codewise looks good, but the description is hard to understand.

File: jobs/cloud_controller_ng/spec (hunk 1 above)
Comment (xandroc) [reply jobs/cloud_controller_ng/spec]: Thanks @joaopapereira, updated

File: jobs/cloud_controller_ng/templates/stack_check.sh.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+set +e
+set -x
+
+CC_PACKAGE_DIR="/var/vcap/packages/cloud_controller_ng"
+export BUNDLE_GEMFILE="${CC_PACKAGE_DIR}/cloud_controller_ng/Gemfile"
+export STACKS_YML="/var/vcap/jobs/cloud_controller_ng/config/stacks.yml"
+export NRCONFIG=/var/vcap/jobs/cloud_controller_ng/config/newrelic.yml
+
+function stack_check {
+      bundle exec rake stacks:stack_check
```
Comment (a-b) [inline jobs/cloud_controller_ng/templates/stack_check.sh.erb:12]: suggestion: this indentation looks inconsistent

File: jobs/cloud_controller_ng/templates/stack_check.sh.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set +e
```
Comment (Samze) [inline jobs/cloud_controller_ng/templates/stack_check.sh.erb:3]: **suggestion** Could we move `set +e` closer to where we don't want to exit, which is wrapped around the rake task?

Comment (xandroc) [reply jobs/cloud_controller_ng/templates/stack_check.sh.erb:3]: updating

Comment (xandroc) [reply jobs/cloud_controller_ng/templates/stack_check.sh.erb]: @Samze updated from offline conversation, should be the same result

```
#!/usr/bin/env bash

set -ex

CC_PACKAGE_DIR="/var/vcap/packages/cloud_controller_ng"
export BUNDLE_GEMFILE="${CC_PACKAGE_DIR}/cloud_controller_ng/Gemfile"
export STACKS_YML="/var/vcap/jobs/cloud_controller_ng/config/stacks.yml"
export NRCONFIG=/var/vcap/jobs/cloud_controller_ng/config/newrelic.yml

function stack_check {
  bundle exec rake stacks:stack_check
}

function main {
  stack_check
}

main

exit 0
```

**Review by Samze — APPROVED** [review]

**Review by a-b — APPROVED** [review]
LGTM

**Review by joaopapereira — APPROVED** [review]
LGTM

**Review by Samze — APPROVED** [review]

**Pull request merged**